--- x/qtbase/src/plugins/platforms/cocoa/qprintengine_mac_p.h
+++ y/qtbase/src/plugins/platforms/cocoa/qprintengine_mac_p.h
@@ -52,6 +52,7 @@
 //

 #include <QtCore/qglobal.h>
+#include <qpa/qplatformprintdevice.h>

 #ifndef QT_NO_PRINTER

--- x/qtbase/src/plugins/plugins.pro
+++ y/qtbase/src/plugins/plugins.pro
@@ -9,6 +9,3 @@ qtHaveModule(gui) {
     !android:qtConfig(library): SUBDIRS *= generic
 }
 qtHaveModule(widgets): SUBDIRS += styles
-
-!winrt:qtHaveModule(printsupport): \
-    SUBDIRS += printsupport
