--- old/qtbase/configure
+++ new/qtbase/configure
@@ -2846,7 +2846,7 @@
         # with the system. We use 'xcrun' to check the clang version that's part of
         # the Xcode installation.
         XCRUN=`/usr/bin/xcrun -sdk macosx clang -v 2>&1`
-        CLANGVERSION=`echo "$XCRUN" | sed -n 's/.*version \([0-9]\).*/\1/p'`
+        CLANGVERSION=`echo "$XCRUN" | sed -n 's/.*version \([0-9]*\).*/\1/p'`
         expr "$CLANGVERSION" : '[0-9]' > /dev/null || { echo "Unable to determine CLANG version from output of xcrun: $XCRUN" ; exit 2 ; }
         if [ "$CLANGVERSION" -ge 3 ]; then
             PLATFORM=macx-clang
