From 3e5fd3fb56bc9ff03beb535979e33dcf83fe1f70 Mon Sep 17 00:00:00 2001
From: Cory Fields <cory-nospam-@coryfields.com>
Date: Thu, 8 May 2014 12:39:42 -0400
Subject: [PATCH] dmg: remove libcrypto dependency

---
 dmg/CMakeLists.txt | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/dmg/CMakeLists.txt b/dmg/CMakeLists.txt
index eec62d6..3969f64 100644
--- a/dmg/CMakeLists.txt
+++ b/dmg/CMakeLists.txt
@@ -1,12 +1,5 @@
-INCLUDE(FindOpenSSL)
 INCLUDE(FindZLIB)
 
-FIND_LIBRARY(CRYPTO_LIBRARIES crypto
-      PATHS
-      /usr/lib
-      /usr/local/lib
-   )
-
 IF(NOT ZLIB_FOUND)
 	message(FATAL_ERROR "zlib is required for dmg!")
 ENDIF(NOT ZLIB_FOUND)
@@ -18,15 +11,6 @@ link_directories(${PROJECT_BINARY_DIR}/common ${PROJECT_BINARY_DIR}/hfs)
 
 add_library(dmg adc.c base64.c checksum.c dmgfile.c dmglib.c filevault.c io.c partition.c resources.c udif.c)
 
-IF(OPENSSL_FOUND)
-	add_definitions(-DHAVE_CRYPT)
-	include_directories(${OPENSSL_INCLUDE_DIR})
-	target_link_libraries(dmg ${CRYPTO_LIBRARIES})
-	IF(WIN32)
-		TARGET_LINK_LIBRARIES(dmg gdi32)
-	ENDIF(WIN32)
-ENDIF(OPENSSL_FOUND)
-
 target_link_libraries(dmg common hfs z)
 
 add_executable(dmg-bin dmg.c)
-- 
2.22.0

