diff --git a/bin/make-base-vm b/bin/make-base-vm
--- a/bin/make-base-vm
+++ b/bin/make-base-vm
@@ -69,6 +69,7 @@ if [ $ARCH = "amd64" -a $SUITE = "hardy" ]; then
 fi
 
 addpkg=openssh-server,pciutils,build-essential,git-core,subversion
+removepkg=cron
 
 if [ -e $OUT.qcow2 ]; then
   echo $OUT.qcow2 already exists, please remove it first
@@ -76,7 +77,7 @@ if [ -e $OUT.qcow2 ]; then
 fi
 
 rm -rf $OUT
-sudo vmbuilder kvm ubuntu --arch=$ARCH --suite=$SUITE --addpkg=$addpkg --ssh-key=var/id_dsa.pub --ssh-user-key=var/id_dsa.pub --mirror=$MIRROR --security-mirror=$SECURITY_MIRROR --dest=$OUT --flavour=$FLAVOUR --firstboot=`pwd`/target-bin/bootstrap-fixup
+sudo vmbuilder kvm ubuntu --arch=$ARCH --suite=$SUITE --addpkg=$addpkg --removepkg=$removepkg --ssh-key=var/id_dsa.pub --ssh-user-key=var/id_dsa.pub --mirror=$MIRROR --security-mirror=$SECURITY_MIRROR --dest=$OUT --flavour=$FLAVOUR --firstboot=`pwd`/target-bin/bootstrap-fixup
 mv $OUT/*.qcow2 $OUT.qcow2
 rm -rf $OUT
 
