Description: Use system JSON Spirit library
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2012-07-05
--- a/src/bitcoinrpc.h
+++ b/src/bitcoinrpc.h
@@ -9,9 +9,7 @@
 #include <string>
 #include <map>
 
-#include "json/json_spirit_reader_template.h"
-#include "json/json_spirit_writer_template.h"
-#include "json/json_spirit_utils.h"
+#include <json_spirit.h>
 
 void ThreadRPCServer(void* parg);
 int CommandLineRPC(int argc, char *argv[]);
--- a/src/makefile.unix
+++ b/src/makefile.unix
@@ -31,6 +31,7 @@ LIBS += \
    -l boost_thread$(BOOST_LIB_SUFFIX) \
    -l db_cxx$(BDB_LIB_SUFFIX) \
    -l ssl \
+   -l json_spirit \
    -l crypto
 
 ifndef USE_UPNP
--- a/src/rpcdump.cpp
+++ b/src/rpcdump.cpp
@@ -8,9 +8,7 @@
 
 #include <boost/lexical_cast.hpp>
 
-#include "json/json_spirit_reader_template.h"
-#include "json/json_spirit_writer_template.h"
-#include "json/json_spirit_utils.h"
+#include <json_spirit.h>
 
 #define printf OutputDebugStringF
 
--- a/src/test/rpc_tests.cpp
+++ b/src/test/rpc_tests.cpp
@@ -1,6 +1,8 @@
 #include <boost/test/unit_test.hpp>
 #include <boost/foreach.hpp>
 
+#include <json_spirit.h>
+
 #include "base58.h"
 #include "util.h"
 #include "bitcoinrpc.h"
--- a/src/test/script_tests.cpp
+++ b/src/test/script_tests.cpp
@@ -8,9 +8,7 @@
 #include <boost/foreach.hpp>
 #include <boost/preprocessor/stringize.hpp>
 #include <boost/test/unit_test.hpp>
-#include "json/json_spirit_reader_template.h"
-#include "json/json_spirit_writer_template.h"
-#include "json/json_spirit_utils.h"
+#include <json_spirit.h>
 
 #include "main.h"
 #include "wallet.h"
