Description: Use system JSON Spirit library
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2012-07-05
--- a/src/bitcoinrpc.cpp
+++ b/src/bitcoinrpc.cpp
@@ -20,9 +20,7 @@
 #include <boost/filesystem/fstream.hpp>
 typedef boost::asio::ssl::stream<boost::asio::ip::tcp::socket> SSLStream;
 #endif
-#include "json/json_spirit_reader_template.h"
-#include "json/json_spirit_writer_template.h"
-#include "json/json_spirit_utils.h"
+#include <json_spirit.h>
 #define printf OutputDebugStringF
 // MinGW 3.4.5 gets "fatal error: had to relocate PCH" if the json headers are
 // precompiled in headers.h.  The problem might be when the pch file goes over
--- a/src/makefile.unix
+++ b/src/makefile.unix
@@ -29,6 +29,7 @@ LIBS += \
    -l boost_thread$(BOOST_LIB_SUFFIX) \
    -l db_cxx$(BDB_LIB_SUFFIX) \
    -l ssl \
+   -l json_spirit \
    -l crypto
 
 ifndef USE_UPNP
--- a/src/rpcdump.cpp
+++ b/src/rpcdump.cpp
@@ -15,9 +15,8 @@
 // typedef boost::asio::ssl::stream<boost::asio::ip::tcp::socket> SSLStream;
 // #endif
 // #include <boost/xpressive/xpressive_dynamic.hpp>
-#include "json/json_spirit_reader_template.h"
-#include "json/json_spirit_writer_template.h"
-#include "json/json_spirit_utils.h"
+
+#include <json_spirit.h>
 
 #define printf OutputDebugStringF
 
--- a/src/test/rpc_tests.cpp
+++ b/src/test/rpc_tests.cpp
@@ -1,11 +1,10 @@
 #include <boost/test/unit_test.hpp>
 #include <boost/foreach.hpp>
 
+#include <json_spirit.h>
+
 #include "base58.h"
 #include "util.h"
-#include "json/json_spirit_reader_template.h"
-#include "json/json_spirit_writer_template.h"
-#include "json/json_spirit_utils.h"
 
 using namespace std;
 using namespace json_spirit;
