diff -ruN miniupnpc-2.2.2/Makefile miniupnpc-2.2.2.new/Makefile
--- miniupnpc-2.2.2/Makefile	2020-11-27 18:25:02.000000000 +0000
+++ miniupnpc-2.2.2.new/Makefile	2024-01-23 20:58:08.387188527 +0000
@@ -298,11 +298,7 @@
 	makedepend -Y -- $(CFLAGS) -- $(SRCS) 2>/dev/null
 
 $(LIBRARY):	$(LIBOBJS)
-ifneq (, $(findstring darwin, $(OS)))
-	$(LIBTOOL) -static -o $@ $?
-else
 	$(AR) crs $@ $?
-endif
 
 $(SHAREDLIBRARY):	$(LIBOBJS)
 ifneq (, $(findstring darwin, $(OS)))
